### PR TITLE
Add swagger docs and client side pages

### PR DESCRIPTION
**Pull request by kingsley9**

### CODE REVIEW

File: AuctionServiceApp/src/main/java/com/EECS4413/AuctionServiceApp/client/CatalogueServiceClient.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.EECS4413.AuctionServiceApp.client;
+
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.*;
+
+import com.EECS4413.AuctionServiceApp.dto.ItemDTO;
+
+@FeignClient(name = "catalogue-service", url = "${catalogue.service.url}")
```
Comment (kingsley9) [inline AuctionServiceApp/src/main/java/com/EECS4413/AuctionServiceApp/client/CatalogueServiceClient.java:8]: Added Feign clients to interact with the other services (Catalogue Service). https://docs.spring.io/spring-cloud-openfeign/docs/current/reference/html/

Comment (DaoudAli) [reply AuctionServiceApp/src/main/java/com/EECS4413/AuctionServiceApp/client/CatalogueServiceClient.java:8]: What's benefit of using this?

Comment (DaoudAli) [reply AuctionServiceApp/src/main/java/com/EECS4413/AuctionServiceApp/client/CatalogueServiceClient.java:8]: @kingsley9

**Review by DaoudAli — APPROVED** [review]

File: AuctionServiceApp/src/main/java/com/EECS4413/AuctionServiceApp/client/CatalogueServiceClient.java (hunk 1 above)
Comment (kingsley9) [reply AuctionServiceApp/src/main/java/com/EECS4413/AuctionServiceApp/client/CatalogueServiceClient.java:8]: Makes it easier to perform inter service communication cause there's less boilerplate code to implement than if we were using rest templates, which is the other option i've found for making api calls within services.

Comment (DaoudAli) [reply AuctionServiceApp/src/main/java/com/EECS4413/AuctionServiceApp/client/CatalogueServiceClient.java:8]: @kingsley9 approved

**Pull request merged**